//
// Copyright 2018 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package hello_world;

import "asylo/enclave.proto";

// Extension to EnclaveInput for use with HelloWorld enclave.

message IoTData {
  optional string MId = 1;
  optional string IDur = 2;
  optional int64 IEnd = 3;
  optional double PAvg = 4;
  optional double EIn = 5;
  optional double EOut = 6;
}

message rawData {
  optional string hash = 1;
  optional string signature = 2;
  repeated IoTData data = 3; 
}

message toClient {
  optional string signature = 1;
  optional string MId = 2;
  optional double PAvg = 3;
  optional int64 IEnd = 4;
}

message ProcessedOutput {
  repeated string signature = 1;
  repeated string MId = 2;
  repeated double PAvg = 3;
  repeated int64 IEnd = 4;
}


extend asylo.EnclaveInput {
  optional rawData parsed_input = 8086;
  optional toClient data_out = 8088;
}

extend asylo.EnclaveOutput {
  optional ProcessedOutput enclave_output_hello = 8087;
}
